💄 style: add feature flag to show model name instead of id

khanhduyvt0101 opened the pull request:
#### 💻 变更类型 | Change Type

<!-- For change type, change [ ] to [x]. -->

- [ ] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [x] 💄 style
- [ ] 👷 build
- [ ] ⚡️ perf
- [ ] 📝 docs
- [ ] 🔨 chore

#### 🔀 变更说明 | Description of Change

This adds a feature flag `model_name` that can be turned on to show model name instead of model id in UI.

khanhduyvt0101 committed:
add sessionToken for bedrock model

khanhduyvt0101 committed:
Support google model list

khanhduyvt0101 committed:
update relate sessionToken info

khanhduyvt0101 committed:
update description and revert gg model list

khanhduyvt0101 committed:
update description

khanhduyvt0101 committed:
merge upstream

khanhduyvt0101 committed:
change session token is a custom field

khanhduyvt0101 committed:
Merge remote-tracking branch 'upstream/main'

khanhduyvt0101 committed:
support google model list

khanhduyvt0101 committed:
Merge remote-tracking branch 'upstream/main'

khanhduyvt0101 committed:
Merge remote-tracking branch 'upstream/main'

khanhduyvt0101 committed:
Merge remote-tracking branch 'upstream/main'

khanhduyvt0101 committed:
Show AI model as name instead of code

khanhduyvt0101 committed:
fix

arvinxx commented:
Actually I perfer to use code for better understanding what's the model id underhood instead of model name.

because model name can be customized but model id should be consistent

khanhduyvt0101 committed:
fix

khanhduyvt0101 committed:
fix

khanhduyvt0101 commented:
> Actually I perfer to use code for better understanding what's the model id underhood instead of model name.
> 
> because model name can be customized but model id should be consistent

I converted into a feature flag, instead. The default remains the same. If user prefers, they can turn on the feature flag to enable this customization. What do you think @arvinxx?

khanhduyvt0101 committed:
fix

arvinxx commented:
> I converted into a feature flag, instead. The default remains the same. If user prefers, they can turn on the feature flag to enable this customization. 

Yeah, a feature flag is a better one,I'm glad to merge it. But it need test to cover the implement at first

khanhduyvt0101 committed:
update UT

khanhduyvt0101 commented:
> > I converted into a feature flag, instead. The default remains the same. If user prefers, they can turn on the feature flag to enable this customization.
> 
> Yeah, a feature flag is a better one,I'm glad to merge it. But it need test to cover the implement at first

I have updated UT to meet coverage standards, please merge it @arvinxx

khanhduyvt0101 commented:
Hi @arvinxx , I just updated base on your feedback

arvinxx commented:
@khanhduyvt0101 hi, plz rebase the code

khanhduyvt0101 committed:
Merge remote-tracking branch 'upstream/main' into show-model-name

khanhduyvt0101 commented:
> @khanhduyvt0101 hi, plz rebase the code

I rebased base on your request @arvinxx

arvinxx commented on src/components/ModelTag/index.tsx:
```diff
@@ -0,0 +1,35 @@
+import { ModelTag as DefaultModelTag, ModelIcon, ModelTagProps } from '@lobehub/icons';
```
add a 'use client'

arvinxx commented on src/components/ModelTag/index.tsx:
```diff
@@ -0,0 +1,35 @@
+
+  let modelCard;
+
+  for (const provider of enabledList) {
```
If you want to display the model name, it's better to have a `provider` props to know the enabled provider. This lookup way will have confilct as the order.

Maybe a optional props with provider,if have this value to just get the model card, or use this lookup way.

arvinxx commented on src/config/featureFlags/schema.ts:
```diff
@@ -20,7 +20,7 @@ export const FeatureFlagsSchema = z.object({
   speech_to_text: z.boolean().optional(),
   token_counter: z.boolean().optional(),
-
+  model_name: z.boolean().optional(),
```
`model_name` is not clear enough for this feature. 

I perfer to `model_tag_display_mode` and with an enum. 

The default value should be `model_id`, and your perference is `model_name` . Maybe we can have another value for `provider/model_id` to show `openai/gpt-4o`

phuctm97 replied:
But feature flags only support boolean values now, doesn't it?

phuctm97 replied:
This docs doesn't mention how to set a enum value for a feature flag: https://lobehub.com/docs/self-hosting/advanced/feature-flags

arvinxx replied:
> But feature flags only support boolean values now, doesn't it?

it seems yes. But we can extend it just like the sytnax of `OPENAI_MODEL_LIST` : https://lobehub.com/docs/self-hosting/advanced/model-list. 

`+key=value`

arvinxx commented on src/config/featureFlags/schema.ts:
```diff
@@ -100,6 +101,7 @@ export const mapFeatureFlagsEnvToState = (config: IFeatureFlags) => {
 
     showMarket: config.market,
     enableSTT: config.speech_to_text,
+    useModelName: config.model_name,
```
key can be `modelTagUseModelName`, value is `config.model_tag_display_mode === 'model_name'`

khanhduyvt0101 committed:
Merge remote-tracking branch 'upstream/main' into show-model-name

khanhduyvt0101 committed:
fix

khanhduyvt0101 committed:
fix UT

khanhduyvt0101 commented:
I just updated base on your review @arvinxx

khanhduyvt0101 committed:
Merge branch 'main' into show-model-name

khanhduyvt0101 committed:
Merge branch 'main' into show-model-name

arvinxx commented:
@khanhduyvt0101 hi, please rebase to main branch and I will merge it.

khanhduyvt0101 committed:
Merge remote-tracking branch 'upstream/main' into show-model-name

khanhduyvt0101 committed:
merge main

khanhduyvt0101 commented:
I rebased @arvinxx

arvinxx commented on docs/self-hosting/advanced/auth/next-auth/casdoor.mdx:
````diff
@@ -50,7 +50,7 @@ There are also some optional fields that can enhance user experience:
 
   ```html
-  <style>
+  <Steps>
````
please keep the previous style tag

arvinxx commented on docs/self-hosting/advanced/auth/next-auth/casdoor.zh-CN.mdx:
````diff
@@ -48,7 +48,7 @@ tags:
 
   ```html
-  <style>
+  <Steps>
````
here the same

arvinxx commented on docs/self-hosting/server-database.mdx:
```diff
@@ -140,5 +140,3 @@ For detailed configuration guidelines on S3, please refer to [S3 Object Storage]
 
 The above is a detailed explanation of configuring LobeChat with a server-side database. You can configure it according to your actual situation and then choose a deployment platform that suits you to start deployment:
-
-<PlatformCards urlPrefix={'server-database'} />
```
revert the change

arvinxx reviewed (commented):
it's OK with left. please update as comments
